packages/market-matcher/src/CertificateService.ts

```diff
@@ -49,10 +54,31 @@ export class CertificateService {
     }
 
     public async matchDemand(certificate: Certificate.Entity, demand: Demand.IDemand) {
-        this.logger.info(`Matched certificate #${certificate.id} to demand #${demand.id}`);
+        if (await this.isAlreadyTransferred(certificate, demand.demandOwner)) {
+            return;
+        }
+
         this.logger.debug(
             `Transferring certificate to ${demand.demandOwner} with account ${this.config.blockchainProperties.activeUser.address}`
         );
+
         await certificate.transferFrom(demand.demandOwner);
     }
+
+    private async isAlreadyTransferred(certificate: Certificate.Entity, owner: string) {
+        const syncedCertificate = await certificate.sync();
+
+        this.logger.verbose(
+            `isAlreadyTransferred: #${syncedCertificate.id} owned by ${syncedCertificate.owner}`
+        );
+
+        if (certificate.owner === owner) {
```

Comment:
I recently fixed bunch of problems where addresses were the same, but different case. Is toLowerCase needed here?